scripts/linters/pylint_extensions.py

```diff
@@ -1960,6 +2059,427 @@ def process_tokens(self, tokens):
+        #   def func(<argument-section>) -> <return-section>:
+        #       <outside function signature block>.
+        if token_type == tokenize.NAME:
+            if token == 'def':
+                is_function_signature_block = False
+                func_def_start_line = line_num
+                outside_args_section = False
+
+        if token_type == tokenize.OP:
+            if token == '->':
+                outside_args_section = True
+                args_section_end_line_num = line_num
+            if outside_args_section and token == ':':
+                is_function_signature_block = True
+
+        # Checking if exceptional_type is present in function definition or not.
+        if token_type == tokenize.NAME and token == exceptional_type:
+            if not outside_args_section:
+                type_present_inside_arg_section = True
+            elif (
+                outside_args_section and
+                args_section_end_line_num == line_num
+            ):
+                type_present_inside_return_section = True
+
+        if (
+            type_present_inside_arg_section or
+            type_present_inside_return_section
+        ):
+            type_present_in_function_signature = True
+
+        if is_function_signature_block:
+            if type_present_in_function_signature:
+                if (
+                    type_comment_present and
+                    func_def_start_line <= (
+                        type_comment_line_num +
+                        ALLOWED_LINES_OF_GAP_IN_COMMENT
+                    )
+                ):
+                    type_comment_present = False
+                else:
+                    self._add_exceptional_type_error_message(
+                        exceptional_type,
+                        func_def_start_line,
+                        node
+                    )
+
+                type_present_in_function_signature = False
+                type_present_inside_arg_section = False
+                type_present_inside_return_section = False
+            if token_type == tokenize.NAME and token == exceptional_type:
+                if exceptional_type == 'object':
+                    # Excluding the case when object is called:
+                    # Eg: var = object()
+                    if 'object()' in line:
+                        return
+                # Passing those cases where Any is imported.
+                if exceptional_type == 'Any':
+                    if (
+                        import_status_dict['single_line_import'] and
+                        import_status_dict['import_line_num'] == line_num
+                    ):
+                        return
+                    elif import_status_dict['inside_multi_line_import_scope']:
+                        return
+                # Checking if comment for exceptional_type is present and it's
+                # with in the range (minimum 10 line of gaps).
```

Comment:
Isn't `ALLOWED_LINES_OF_GAP_IN_COMMENT` a maximum, and isn't it 15 not 10?

Reply:
FIxed